lib/CL/devices/pthread/pthread.c

```diff
@@ -117,6 +113,10 @@ struct data {
   mem_regions_management* mem_regions;
 #endif
 
+  /* Lock for command list related operations */
+  pthread_mutex_t cq_lock __attribute__ ((aligned (CACHELINE_SIZE)));
```

Comment:
Also not helping is the amount of volatiles (though i understand why some are required), since every volatile means fetching from memory = fighting other driver threads for cachelines :) so perhaps surprising but yes alignment does the most speedup here.